rfcs/0192-min-capacity-ensures-workers-do-not-get-unnecessarily-killed.md

```diff
@@ -0,0 +1,229 @@
+- **Azure**: [Worker ID maps to VM name](https://github.com/taskcluster/taskcluster/blob/d7fbf0ee9e0d93e079cc7ff069eaceecfd7d29ec/services/worker-manager/src/providers/azure/index.js#L302), uses [VM deletion API](https://github.com/taskcluster/taskcluster/blob/d7fbf0ee9e0d93e079cc7ff069eaceecfd7d29ec/services/worker-manager/src/providers/azure/index.js#L1249-L1254)
+
+**Implementation:**
+Worker-manager has existing capability to forcefully terminate workers by updating the database to mark workers as STOPPING state and then making direct cloud provider API calls to terminate the instances. This process works across all supported cloud providers (GCP, AWS, Azure) using their respective termination APIs.
+
+**Implementation References:**
+- [Worker.states.STOPPING definition](https://github.com/taskcluster/taskcluster/blob/d7fbf0ee9e0d93e079cc7ff069eaceecfd7d29ec/services/worker-manager/src/data.js#L858-L863)
+- [AWS removeWorker implementation](https://github.com/taskcluster/taskcluster/blob/d7fbf0ee9e0d93e079cc7ff069eaceecfd7d29ec/services/worker-manager/src/providers/aws.js#L382-L418)
+- [Google removeWorker implementation](https://github.com/taskcluster/taskcluster/blob/d7fbf0ee9e0d93e079cc7ff069eaceecfd7d29ec/services/worker-manager/src/providers/google.js#L168-L192)
+- [Azure removeWorker implementation](https://github.com/taskcluster/taskcluster/blob/d7fbf0ee9e0d93e079cc7ff069eaceecfd7d29ec/services/worker-manager/src/providers/azure/index.js#L1221-L1233)
+
+**Trade-offs:**
+- ✅ **Immediate capacity response**: `minCapacity` changes take effect in ~30 seconds
+- ❌ **Cache loss**: Defeats primary optimization goal when terminating active workers
+- ⚠️ **Selective termination**: Prioritize terminating idle workers when possible
+
+#### 6. Enhanced Health Monitoring
+
+**MinCapacity Worker Health Checks:**
+- More aggressive health monitoring for `minCapacity` workers
+- Faster detection of unresponsive `minCapacity` workers
+- Automatic detection and fallback if `minCapacity` workers cause issues
+
+#### 7. Monitoring and Alerting
+
+**Implementation Location:**
+New metrics will be added to [services/worker-manager/src/monitor.js](https://github.com/taskcluster/taskcluster/blob/d7fbf0ee9e0d93e079cc7ff069eaceecfd7d29ec/services/worker-manager/src/monitor.js#L377) following the existing pattern established in [lines 246-377](https://github.com/taskcluster/taskcluster/blob/d7fbf0ee9e0d93e079cc7ff069eaceecfd7d29ec/services/worker-manager/src/monitor.js#L246-L377). These metrics will be displayed in the Taskcluster UI Status Dashboard for human operators and exposed via Prometheus `/metrics` endpoint for external monitoring systems to create custom dashboards and alerting.
+
+**New Prometheus Metrics:**
```

Comment:
Do we really need all of those metrics? 
I feel like existing metrics are enough, and in case we stumble upon some errors we'd be looking in longs anyway. 
We have `*_queue_running_workers` to count how many active workers there are

Reply:
New metrics removed in favor of existing ones. Thanks!